app/models/application.rb

```diff
@@ -20,13 +20,14 @@ class Application < ApplicationRecord
 
   default_scope { order("name ASC") }
 
-  def latest_deploy_to_each_environment
-    return @latest_deploy_to_each_environment unless @latest_deploy_to_each_environment.nil?
+  ENVIRONMENTS_ORDER = %w[integration staging production].freeze
 
-    environments = deployments.select("DISTINCT environment").map(&:environment)
-    @latest_deploy_to_each_environment = environments.index_with do |environment|
+  def latest_deploy_to_each_environment
+    envs = deployed_to_ec2? ? ENVIRONMENTS_ORDER : ENVIRONMENTS_ORDER.map { |env| "#{env} EKS" }
```

Comment:
If I'm reading this right, there'll also be a small functional change in that app status pages used to show _all_ environments (training, preview, etc, but also EKS vs non EKS, so integration, integration EKS, staging, staging EKS, etc).

Whereas now, if the app is deployed to EKS, we will _only_ show the EKS environments (integration EKS, staging EKS, production EKS), and stop showing the non-EKS ones.

This is fine - but probably worth mentioning this in the commit message and PR description 👍

Reply:
Thank you yeah, it came about when the consensus became that non-main environments shouldn't display, so it now displays the environments relevant to the application. (Where it is being deployed.) I have now noted it in the PR and the commit message.